### PR TITLE
feat(eventing): add host events

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added host events.

### CODE REVIEW

File: io-engine/src/eventing/host_events.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+}
+
+impl EventWithMeta for NvmfController {
+    fn event(
+        &self,
+        event_action: EventAction,
+        meta: EventMeta,
+    ) -> EventMessage {
+        let mut event_meta = meta.clone();
```
Comment (tiagolobocastro) [inline io-engine/src/eventing/host_events.rs]: Do we need to clone, can't we just mutate meta?

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/eventing/host_events.rs (hunk 1 above)
Comment (datacore-vvarakantham) [reply io-engine/src/eventing/host_events.rs]: Yes we can mutate meta. Added the change now.

**Review by chriswldenyer — APPROVED** [review]
looks reasonable

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**